infrastructure/src/main/java/org/wso2/testgrid/infrastructure/providers/AWSProvider.java

```diff
@@ -270,4 +279,25 @@ private boolean doRelease(InfrastructureConfig infrastructureConfig, String stac
         return cfCompatibleParameters;
     }
 
+    private String getFailedStackEvents(String stackName, AmazonCloudFormation cloudFormation) {
+        DescribeStackEventsRequest describeStackEventsRequest = new DescribeStackEventsRequest()
+                .withStackName(stackName);
+        DescribeStackEventsResult describeStackEventsResult = cloudFormation.
+                describeStackEvents(describeStackEventsRequest);
+        StringBuilder stringBuilder = new StringBuilder();
+        for (StackEvent stackEvent: describeStackEventsResult.getStackEvents()) {
+            if (stackEvent.getResourceStatus().equals(CREATE_FAILED)) {
+                stringBuilder.append(StringUtil.concatStrings(
+                        "Status: ", stackEvent.getResourceStatus(), ", "));
+                stringBuilder.append(StringUtil.concatStrings(
+                        "Resource Type: ", stackEvent.getResourceType(), ", "));
+                stringBuilder.append(StringUtil.concatStrings(
+                        "Logical ID: ", stackEvent.getLogicalResourceId(), ", "));
+                stringBuilder.append(StringUtil.concatStrings(
+                        "Status Reason: ", stackEvent.getResourceStatusReason()));
```

Comment:
IMO, the keys will get duplicated for each failed event here!
Ideal way here is to print this like a table.. Like the mysql `select` command output table.
Not required for this pr.